mlflow/server/handlers.py

```diff
@@ -799,6 +823,64 @@ def _delete_model_version_tag():
     return _wrap_response(DeleteModelVersionTag.Response())
 
 
+@catch_mlflow_exception
+def _download_artifact(artifact_path):
+    """
+    A request handler for `GET /mlflow-artifacts/artifacts/<artifact_path>` to download an artifact
+    from `artifact_path` (a relative path from the root artifact directory).
+    """
+    basename = posixpath.basename(artifact_path)
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        tmp_path = os.path.join(tmp_dir, basename)
+        artifact_repo = _get_artifact_repo_mlflow_artifacts()
+        artifact_repo._download_file(artifact_path, tmp_path)
+        return send_from_directory(tmp_dir, basename)
```

Comment:
@dbczumar On windows, this code doesn't work:

https://github.com/mlflow/mlflow/runs/4102743134?check_suite_focus=true#step:5:1887

```
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpoug8eouu\\a.txt'
```

This SO question reports the exact same issue:

https://stackoverflow.com/questions/56581717/in-flask-how-can-i-send-a-temporary-file-and-delete-it-after-upload-is-finished?noredirect=1&lq=1

> It looks like `send_file` is asynchronous, so Python tries to delete the temporary directory while another thread is still uploading the file. On Linux, this doesn't happen because only the filenames are deleted, while the file node will still be alive until the other thread finishes uploading the file.

Reply:
This SO answer provides a few workarounds: https://stackoverflow.com/a/24613980

Reply:
"Stream file, then delete" from https://stackoverflow.com/a/24613980/11952869 seems good!

Reply:
Let's try that!

Reply:
The stream-then-delete approach worked on windows!